app/controllers/sitemap_controller.rb

```diff
@@ -30,21 +30,28 @@ def build
 
           xml.url do
             xml.loc(request.base_url + page_location(path))
-            xml.lastmod(metadata.fetch(:date) { DEFAULT_LASTMOD })
+            xml.lastmod(lastmod_date(path, metadata))
             xml.priority(metadata.fetch(:priority)) if metadata.key?(:priority)
           end
         end
 
         OTHER_PATHS.each do |events_path|
           xml.url do
             xml.loc(request.base_url + events_path)
-            xml.lastmod(DEFAULT_LASTMOD)
+            xml.lastmod(lastmod_date(events_path))
           end
         end
       end
     end
   end
 
+  def lastmod_date(path, metadata = nil)
+    page_mod = PageModification.find_by(path: path)
```

Comment:
I think it will be more efficient to return the `metadata&.dig(:date)` if its present, before querying the database if its not